pandas/core/groupby/generic.py

```diff
@@ -1069,13 +1069,37 @@ def nsmallest(
 
     @doc(Series.idxmin.__doc__)
     def idxmin(self, axis: Axis = 0, skipna: bool = True) -> Series:
-        result = self._op_via_apply("idxmin", axis=axis, skipna=skipna)
-        return result
+        axis = self.axis
```

Comment:
you're overriding the axis keyword here, seems like trouble

Reply:
Indeed! It was a careless error brought on by copying, thanks for the reminder